### PR TITLE
Better csv upload

### LINKED ISSUE #279: Flesh out student csv import
Build on #249

### PR DESCRIPTION
**Pull request by seveibar**
fixes #279 
- Admin is delivered error/success messages of each user
- Duplicate users will not be created (admin will be informed)
- Instructors can now be created
- Sample spreadsheet can be downloaded so the format is known

Special bonus feature: Fixed the user deletion (empty pw fails) and added a parameter for true deletion.

Note: Email is outside the scope of this PR and still "in the works" by @leej42 . It will probably be a part of the presave hook and will automatically just work.

### CODE REVIEW

**Comment by kmcnellis** [comment]
Please add a sample csv to the api docs

File: client/app/admin/admin.html [hunk 1]
```diff
@@ -3,8 +3,11 @@
 <div class="container">
   <h4> User Control </h4>
   <form>
-    <div class="btn btn-default btn-lg" ngf-select="uploadUserCSV($file)">Upload students with CSV</div>
+    <div class="btn btn-default btn-lg" ngf-select="uploadUserCSV($file)">Upload users with CSV</div>
```
Comment (kmcnellis) [inline client/app/admin/admin.html:6]: This styling looks bad

File: server/api/user/user.controller.js [hunk 2]
```diff
@@ -57,10 +58,15 @@ export function create(req, res, next) {
...
 
+export function getExampleCSVUpload(req, res, next){
+  res.sendFile(path.join(__dirname,"./example_csv_upload.csv"));
+}
+
 /**
  * Creates users in database from a CSV file
  */
 export function createFromCSVUpload(req, res, next){
+  console.log("Attempting csv upload");
```
Comment (kmcnellis) [inline server/api/user/user.controller.js]: Console.log

File: server/auth/local/index.js [hunk 3]
```diff
@@ -15,9 +18,19 @@ router.post('/', function(req, res, next) {
     if (!user) {
       return res.status(404).json({message: 'Something went wrong, please try again.'});
     }
+
     var profile = user.profile;
-    var token = signToken(user._id, user.role);
-    res.json({ token, profile });
+    User.findById(profile._id).then((user) => {
```
Comment (kmcnellis) [inline server/auth/local/index.js:24]: This should not be part of this PR. Please branch from dev before working on separate feature in the future. Would not be able merge this due to problems in other PR

File: client/app/admin/admin.controller.js [hunk 4]
```diff
@@ -19,7 +19,7 @@ angular.module('venueApp')
           objectKey: '.k',
           arrayKey: '[i]'
       }).success(((response) => {
-        console.log("success", response);
+        $scope.csvResults = response.join("<br/>");
```
Comment (kmcnellis) [inline client/app/admin/admin.controller.js:22]: Should update the list of students when uploading is complete

File: client/app/admin/admin.html (hunk 1 above)
Comment (kmcnellis) [inline client/app/admin/admin.html:6]: Make the text/button smaller, or add margin on bottom

**Comment by kmcnellis** [comment]
Created #293 as the next step

Good once issues are resolved and the dependency on #289 is removed

File: client/app/admin/admin.html (hunk 1 above)
Comment (seveibar) [inline client/app/admin/admin.html:6]: I'm gonna say this is :penguin:? This is a P1

File: client/app/admin/admin.controller.js (hunk 4 above)
Comment (seveibar) [inline client/app/admin/admin.controller.js:22]: good catch

File: server/auth/local/index.js (hunk 3 above)
Comment (seveibar) [inline server/auth/local/index.js:24]: These PRs have an order to them, I suppose this is a bad habit when under time constraint- this portion has already been merged

**Comment by seveibar** [comment]
thanks for the review #289 merge resolves dependency. Good next step. API documentation added.

**Pull request merged**